### ISSUE TITLE
Block signature verification error

### ISSUE THREAD

**Issue by Maxime2** [andrecronje/lachesis#42]
Soon after we run 3 node cluster we see the following error (I post here only head of the error message as the full log file is about 31M of size; the full log is attached in compressed form)

This is the same issue as noted here: https://github.com/andrecronje/lachesis/pull/41

```
Could not load store - creating new
time="2018-10-15T00:27:37Z" level=info msg=Listening addr="192.168.0.5:12000"
Setting root 0x0410AE1E7472DF0C758354569F7725D622566B9E644463C933E244552C9B116C76ACF7F0C01CFE4D3A9E79F055C910ACBF1A42102B2A26E542FB64956A40B96E6B -> [48 120 48 52 49 48 65 69 49 69 55 52 55 50 68 70 48 67 55 53 56 51 53 52 53 54 57 70 55 55 50 53 68 54 50 50 53 54 54 66 57 69 54 52 52 52 54 51 67 57 51 51 69 50 52 52 53 53 50 67 57 66 49 49 54 67 55 54 65 67 70 55 70 48 67 48 49 67 70 69 52 68 51 65 57 69 55 57 70 48 53 53 67 57 49 48 65 67 66 70 49 65 52 50 49 48 50 66 50 65 50 54 69 53 52 50 70 66 54 52 57 53 54 65 52 48 66 57 54 69 54 66 95 114 111 111 116]
Setting root 0x04609B5461F930185E9BBCA696CA6B1022DF9C96F9B1CE87819224560130A5029906691D7EABBE0557C55348B74A046DCAF0108A90AB751CDC99F8797E88330C0F -> [48 120 48 52 54 48 57 66 53 52 54 49 70 57 51 48 49 56 53 69 57 66 66 67 65 54 57 54 67 65 54 66 49 48 50 50 68 70 57 67 57 54 70 57 66 49 67 69 56 55 56 49 57 50 50 52 53 54 48 49 51 48 65 53 48 50 57 57 48 54 54 57 49 68 55 69 65 66 66 69 48 53 53 55 67 53 53 51 52 56 66 55 52 65 48 52 54 68 67 65 70 48 49 48 56 65 57 48 65 66 55 53 49 67 68 67 57 57 70 56 55 57 55 69 56 56 51 51 48 67 48 70 95 114 111 111 116]
Setting root 0x043DFC59DFC928E0108BE1280D6C54F1763B11C64F77CD43AE6A636E5E3A7C1824A122C7893A67AD3635A076B3A2DD6AD00C27CBF7D343E12684CAA34E4344F06F -> [48 120 48 52 51 68 70 67 53 57 68 70 67 57 50 56 69 48 49 48 56 66 69 49 50 56 48 68 54 67 53 52 70 49 55 54 51 66 49 49 67 54 52 70 55 55 67 68 52 51 65 69 54 65 54 51 54 69 53 69 51 65 55 67 49 56 50 52 65 49 50 50 67 55 56 57 51 65 54 55 65 68 51 54 51 53 65 48 55 54 66 51 65 50 68 68 54 65 68 48 48 67 50 55 67 66 70 55 68 51 52 51 69 49 50 54 56 52 67 65 65 51 52 69 52 51 52 52 70 48 54 70 95 114 111 111 116]
PingNodesN::participants:  [0xc00010b5f0 0xc00010b620 0xc00010b530]
PingNodesN::p:  map[0x043DFC59DFC928E0108BE1280D6C54F1763B11C64F77CD43AE6A636E5E3A7C1824A122C7893A67AD3635A076B3A2DD6AD00C27CBF7D343E12684CAA34E4344F06F:0xc00010b5f0 0x0410AE1E7472DF0C758354569F7725D622566B9E644463C933E244552C9B116C76ACF7F0C01CFE4D3A9E79F055C910ACBF1A42102B2A26E542FB64956A40B96E6B:0xc00010b620 0x04609B5461F930185E9BBCA696CA6B1022DF9C96F9B1CE87819224560130A5029906691D7EABBE0557C55348B74A046DCAF0108A90AB751CDC99F8797E88330C0F:0xc00010b530]
time="2018-10-15T00:27:41Z" level=info msg=Dialing target="192.168.0.4:12000" timeout=1s
time="2018-10-15T00:27:41Z" level=warning msg="failed to get parent: %sEvent, Root2561334434, Not Found" id=2561334434
time="2018-10-15T00:27:41Z" level=warning msg="failed to get  other parent: %sKey cannot be empty" id=2561334434
time="2018-10-15T00:27:42Z" level=warning msg="logStats()" consensus_events=0 consensus_transactions=0 events/s=0.00 id=2561334434 last_block_index=-1 last_consensus_round=nil num_peers=3 round_events=0 rounds/s=0.00 state=Gossiping sync_rate=1.00 t/s=0.00 this_id=2561334434 transaction_pool=1400 undetermined_events=1
time="2018-10-15T00:27:49Z" level=info msg="accepted connection" from="192.168.0.3:47636" node="192.168.0.5:12000"
time="2018-10-15T00:27:49Z" level=info msg=Dialing target="192.168.0.3:12000" timeout=1s
time="2018-10-15T00:27:50Z" level=error msg="n.sync(resp.Events)" error="CheckSelfParent: Self-parent not last known event by creator" this_id=2561334434
time="2018-10-15T00:27:51Z" level=info msg="accepted connection" from="192.168.0.4:39612" node="192.168.0.5:12000"
time="2018-10-15T00:27:55Z" level=warning msg="failed to get  other parent: %sKey cannot be empty" id=2561334434
time="2018-10-15T00:27:56Z" level=error msg="n.requestEagerSync(peerAddr, wireEvents)" Error="read tcp 192.168.0.5:59062->192.168.0.3:12000: i/o timeout" this_id=2561334434
time="2018-10-15T00:27:56Z" level=info msg="accepted connection" from="192.168.0.3:49878" node="192.168.0.5:12000"
time="2018-10-15T00:28:01Z" level=info msg=Dialing target="192.168.0.3:12000" timeout=1s
time="2018-10-15T00:28:02Z" level=warning msg="logStats()" consensus_events=0 consensus_transactions=0 events/s=0.00 id=2561334434 last_block_index=-1 last_consensus_round=0 num_peers=3 round_events=0 rounds/s=0.00 state=Gossiping sync_rate=1.00 t/s=0.00 this_id=2561334434 transaction_pool=2500 undetermined_events=18
time="2018-10-15T00:28:05Z" level=warning msg="failed to get  other parent: %sKey cannot be empty" id=2561334434
time="2018-10-15T00:28:05Z" level=warning msg="logStats()" consensus_events=0 consensus_transactions=0 events/s=0.00 id=2561334434 last_block_index=-1 last_consensus_round=0 num_peers=3 round_events=0 rounds/s=0.00 state=Gossiping sync_rate=1.00 t/s=0.00 this_id=2561334434 transaction_pool=1100 undetermined_events=22
time="2018-10-15T00:28:12Z" level=warning msg="failed to get  other parent: %sKey cannot be empty" id=2561334434
time="2018-10-15T00:28:13Z" level=warning msg="logStats()" consensus_events=0 consensus_transactions=0 events/s=0.00 id=2561334434 last_block_index=-1 last_consensus_round=0 num_peers=3 round_events=0 rounds/s=0.00 state=Gossiping sync_rate=1.00 t/s=0.00 this_id=2561334434 transaction_pool=1937 undetermined_events=26
time="2018-10-15T00:28:20Z" level=warning msg="Verifying Block signature. Invalid signature" block="{{0 1 [] [188 139 81 237 76 125 222 190 255 129 71 178 120 249 208 8 210 170 149 238 91 208 164 122 121 65 38 57 247 113 102 147] [[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] [0 0 0 0 0 0 0 0 0 0 0 ...

```
[00-block-signature-err.zip](https://github.com/andrecronje/lachesis/files/2477051/00-block-signature-err.zip)

**Comment by andrecronje** [comment]
Investigating in #41

**Comment by SamuelMarks** [comment]
@andrecronje FYI this bug is present on your current master, so feel free to merge #41.

**Comment by Maxime2** [comment]
The issue has been resolved in the commit 07e61a016bcfe4db38894e3ac0c50bc6e7ace857 , so I am closing this issue.
